astropy/visualization/scripts/fits2bitmap.py

```diff
@@ -71,11 +69,21 @@ def fits2bitmap(filename, ext=0, out_fn=None, scale='linear',
     import matplotlib.cm as cm
     import matplotlib.image as mimg
 
-    hdulist = fits.open(filename)
-    image = hdulist[ext].data
-    hdulist.close()
+    try:
+        ext = int(ext)
+    except ValueError:
+        pass
+
+    image = getdata(filename, ext)
     if out_fn is None:
-        out_fn = filename.replace('.fits', '.png')
+        if filename.endswith('fits.gz'):
+            out_fn = filename.replace('.fits.gz', '.png')
+        elif filename.endswith('fits.bz2'):
+            out_fn = filename.replace('.fits.bz2', '.png')
+        elif filename.endswith('fits.zip'):
+            out_fn = filename.replace('.fits.zip', '.png')
+        else:
+            out_fn = filename.replace('.fits', '.png')
     if cmap not in cm.datad.keys():
```

Comment:
I realize this line was already here, but it should just be `if cmap not in cm.datad`.